astropy/io/misc/hdf5.py

```diff
@@ -58,7 +58,7 @@ def is_hdf5(origin, filepath, fileobj, *args, **kwargs):
         return isinstance(args[0], (h5py.File, h5py.Group, h5py.Dataset))
 
 
-def read_table_hdf5(input, path=None):
+def read_table_hdf5(input, path=None, character_as_bytes=True):
```

Comment:
Do we really need this new option to convert to unicode at the HDF5-reader level?  With the Table unicode-sandwich this should generally be unnecessary.  Have any users requested this functionality?  I would say to just stick to fixing writing, which is actually broken.